bundles/core/src/main/java/com/adobe/cq/commerce/core/cacheinvalidation/internal/InvalidateDispatcherCacheImpl.java

```diff
@@ -39,200 +33,223 @@
+                LOGGER.error("Error processing strategy for key: {}", entry.getKey(), e);
             }
         }
 
-        return Stream.concat(Arrays.stream(invalidateDispatcherPagePaths), Arrays.stream(correspondingPaths))
-            .toArray(String[]::new);
+        LOGGER.debug("Found {} unique paths to invalidate", allPaths.size());
+        return new ArrayList<>(allPaths);
     }
 
-    protected String[] getCorrespondingPageBasedOnEntries(Session session, String storePath, String[] entries, String key)
-        throws CacheInvalidationException {
-        String entryList = formatList(entries, ", ", "'%s'");
-        try {
-            String sqlQuery = invalidateCacheRegistry.getQuery(key, storePath, entryList);
-            if (sqlQuery != null) {
-                return getQueryResult(getSqlQuery(session, sqlQuery));
-            }
-        } catch (Exception e) {
-            throw new CacheInvalidationException("Error getting corresponding page based on entries", e);
-        }
-        return new String[0];
-    }
+    private Set<String> processAttributeStrategy(Map.Entry<String, String[]> entry, Page page,
+        ResourceResolver resourceResolver, String storePath, MagentoGraphqlClient client) {
+        Set<String> paths = new HashSet<>();
+        AttributeStrategies strategies = invalidateCacheRegistry.getAttributeStrategies(entry.getKey());
 
-    protected String[] getPathsToInvalidate(ResourceResolver resourceResolver, Map<String, Object> data,
-        String storePath, String key) {
-        Page page = getPage(resourceResolver, storePath);
-        return invalidateCacheRegistry.getPathsToInvalidate(key, page, resourceResolver, data, storePath);
+        strategies.getStrategies(false).stream()
+            .filter(info -> info.getStrategy() instanceof DispatcherCacheInvalidationStrategy)
+            .forEach(info -> {
+                try {
+                    DispatcherCacheInvalidationStrategy strategy = (DispatcherCacheInvalidationStrategy) info.getStrategy();
+                    List<String> attributeData = new ArrayList<>(Arrays.asList(entry.getValue()));
+
+                    DispatcherCacheInvalidationContext context = new DispatcherCacheInvalidationContextImpl(
+                        page,
+                        resourceResolver,
+                        attributeData,
+                        storePath,
+                        client);
+
+                    List<String> invalidPaths = strategy.getPathsToInvalidate(context);
+                    if (invalidPaths != null) {
+                        paths.addAll(invalidPaths.stream()
+                            .filter(Objects::nonNull)
+                            .filter(path -> !path.trim().isEmpty())
+                            .collect(Collectors.toSet()));
+                    }
+                } catch (Exception e) {
+                    LOGGER.error("Error processing strategy for key: {}", entry.getKey(), e);
```

Comment:
Please use this code here since key is not informative enough about the actual source of the exception:
`LOGGER.error("Exception was thrown by {}: ", info.getStrategy().getClass(), e);`